src/relay/qnn/op/convolution.cc

```diff
@@ -40,6 +40,23 @@ namespace qnn {
 // relay.op.qnn.conv2d
 TVM_REGISTER_NODE_TYPE(QnnConv2DAttrs);
 
+bool QnnConv2DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
+               const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 3);
+  const auto* data = types[0].as<TensorTypeNode>();
```

Comment:
assert on data, weight and param would be useful to check that they are not nullptr. Else it will fail with a silent segfault.